### PR TITLE
Feature/unr 2494 expose additional connection config settings

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
Exposed tcp and udp connection config settings to gdk settings.
Updated to 14.2.1

https://github.com/improbableio/UnrealEngine/pull/308
https://github.com/improbableio/UnrealEngine/pull/309
https://github.com/spatialos/UnrealGDKExampleProject/pull/80
https://github.com/spatialos/UnrealGDKTestGyms/pull/10
https://github.com/spatialos/UnrealGDKThirdPersonShooter/pull/99

#### Documentation
No user-facing documentation yet - that will be done in UNR-2514

#### Primary reviewers
@aleximprobable

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h [hunk 1]
```diff
@@ -156,7 +156,7 @@ class SPATIALGDK_API USpatialGDKSettings : public UObject
 	uint32 MaxDynamicallyAttachedSubobjectsPerClass;
 
 	/** EXPERIMENTAL - This is a stop-gap until we can better define server interest on system entities.
-	Disabling this is not supported in any type of multi-server environment*/
+	Disabling this is not supported in a zoned-worker environment*/
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h]: Does that mean it's fine for Offloading?

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h]: Yes

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h [hunk 2]
```diff
@@ -55,6 +79,9 @@ struct FConnectionConfig
 	Worker_NetworkConnectionType LinkProtocol;
 	Worker_ConnectionParameters ConnectionParams;
 	uint8 TcpMultiplexLevel;
+	uint8 TcpNoDelay;
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h]: It looks like a `bool` to me

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp [hunk 3]
```diff
@@ -185,77 +248,28 @@ void USpatialWorkerConnection::StartDevelopmentAuth(FString DevAuthToken)
 	}
 }
 
-void USpatialWorkerConnection::ConnectToReceptionist(bool bConnectAsClient)
+void USpatialWorkerConnection::ConnectToReceptionist()
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp]: I'll trust that you've refactored `ConnectToReceptionist` and `ConnectToLocator` functions correctly (i.e. didn't check every line there, but I like and approve this change in general 👍 )

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h [hunk 4]
```diff
@@ -18,6 +18,9 @@ struct FConnectionConfig
 		, EnableProtocolLoggingAtStartup(false)
 		, LinkProtocol(WORKER_NETWORK_CONNECTION_TYPE_MODULAR_UDP)
 		, TcpMultiplexLevel(2) // This is a "finger-in-the-air" number.
+		, TcpNoDelay(0)
+		, UdpUpstreamIntervalMS(10)
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h]: I'd rather put `0` and not `10` here, to explicitly state we init it with default values and `PreConnectInit` must be called to read proper values from the Settings.

File: SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp [hunk 5]
```diff
@@ -43,6 +43,13 @@ USpatialGDKSettings::USpatialGDKSettings(const FObjectInitializer& ObjectInitial
 	, WorkerLogLevel(ESettingsWorkerLogVerbosity::Warning)
 	, SpatialDebuggerClassPath(TEXT("/SpatialGDK/SpatialDebugger/BP_SpatialDebugger.BP_SpatialDebugger_C"))
 	, bEnableUnrealLoadBalancer(false)
+	// TODO - UNR 2514 - These defaults are not necessarily optimal - readdress when we have better data
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp]: Where do these values come from? Ideally, they should be equal to the ones we had by default before this PR, but I can't find these.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp (hunk 3 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp]: Your trust is well placed

File: SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp (hunk 5 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp]: From Worker SDK defaults - which aren't exposed.

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h (hunk 4 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h]: Fair enough

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h (hunk 2 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h]: Kept it an uint as that's what the worker config wants

**Pull request merged**